### ISSUE TITLE
last doesn't return last entry in table

### ISSUE THREAD

**Issue by charlemagnelasse** [influxdata/influxdb#20722]
__Steps to reproduce:__
List the minimal actions needed to reproduce the behavior.

1. Create measurement with more with data for more than a day
2. query the data
3. get the last entry

Here some output:

```
$ influx query --org test 
from(bucket:"test/autogen")
        |> range(start:-15h)
        |> filter(fn: (r) =>
                r._measurement == "foobaricatio" and
                r._field == "x")
        |> last()
Result: _result
Table: keys: [_start, _stop, _field, _measurement, serial]
                   _start:time                      _stop:time           _field:string     _measurement:string           serial:string                  _value:float                      _time:time  
------------------------------  ------------------------------  ----------------------  ----------------------  ----------------------  ----------------------------  ------------------------------  
2021-02-07T23:40:11.725006162Z  2021-02-08T14:40:11.725006162Z                       x            foobaricatio                12345678                        -0.055  2021-02-07T23:59:50.131313920Z  

$ influx query --org test 
from(bucket:"test/autogen")
        |> range(start:-15h)
        |> filter(fn: (r) =>
                r._measurement == "foobaricatio" and
                r._field == "x")
Result: _result
Table: keys: [_start, _stop, _field, _measurement, serial]
                   _start:time                      _stop:time           _field:string     _measurement:string           serial:string                  _value:float                      _time:time  
------------------------------  ------------------------------  ----------------------  ----------------------  ----------------------  ----------------------------  ------------------------------  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                         -0.05  2021-02-07T23:40:29.176546048Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                         -0.05  2021-02-07T23:40:29.188487936Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                        -0.065  2021-02-07T23:40:59.173333760Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                        -0.055  2021-02-07T23:41:28.858528000Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                        -0.065  2021-02-07T23:41:58.589042944Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                        -0.065  2021-02-07T23:42:28.378181888Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                        -0.055  2021-02-07T23:42:58.133071872Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                         -0.06  2021-02-07T23:43:27.844079872Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                        -0.055  2021-02-07T23:43:57.683383040Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                        -0.065  2021-02-07T23:44:27.389230080Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                        -0.055  2021-02-07T23:44:57.211719936Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                         -0.05  2021-02-07T23:45:26.953754880Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                         -0.05  2021-02-07T23:45:56.695632896Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                         -0.06  2021-02-07T23:46:26.462795008Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                         -0.05  2021-02-07T23:46:56.224004864Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                         -0.06  2021-02-07T23:47:26.039407104Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                         -0.06  2021-02-07T23:47:55.748610048Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                         -0.06  2021-02-07T23:48:25.611627008Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                        -0.055  2021-02-07T23:48:55.294322944Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                        -0.055  2021-02-07T23:49:25.017849088Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                        -0.055  2021-02-07T23:49:54.789655040Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                        -0.065  2021-02-07T23:50:24.553970944Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                        -0.055  2021-02-07T23:50:54.341972992Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                        -0.065  2021-02-07T23:51:24.061400064Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                        -0.055  2021-02-07T23:51:53.870439936Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                        -0.055  2021-02-07T23:52:23.605427968Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                        -0.055  2021-02-07T23:52:53.377667840Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                        -0.055  2021-02-07T23:53:23.147703040Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                         -0.06  2021-02-07T23:53:52.889359104Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                        -0.065  2021-02-07T23:54:22.640892928Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                        -0.045  2021-02-07T23:54:52.382606080Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                        -0.065  2021-02-07T23:55:22.247430144Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                         -0.05  2021-02-07T23:55:51.943030016Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                         -0.05  2021-02-07T23:56:21.650744832Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                         -0.05  2021-02-07T23:56:21.691958016Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                         -0.05  2021-02-07T23:56:51.535497984Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                        -0.065  2021-02-07T23:57:21.329739008Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                         -0.06  2021-02-07T23:57:51.116591104Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                         -0.06  2021-02-07T23:58:20.859128832Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                         -0.05  2021-02-07T23:58:50.633061888Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                        -0.055  2021-02-07T23:59:20.329579008Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                        -0.055  2021-02-07T23:59:50.131313920Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                         -0.06  2021-02-08T00:00:19.883734016Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                         -0.05  2021-02-08T00:00:49.658305024Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                        -0.055  2021-02-08T00:01:19.394260992Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                        -0.055  2021-02-08T00:01:49.141611008Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                        -0.055  2021-02-08T00:02:18.893554944Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                         -0.06  2021-02-08T00:02:48.587849984Z  
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                         -0.05  2021-02-08T00:03:18.429417984Z  
[...]
2021-02-07T23:40:22.732894532Z  2021-02-08T14:40:22.732894532Z                       x            foobaricatio                12345678                         -0.06  2021-02-08T14:39:53.035065856Z  

```

__Expected behavior:__

I get the last written entry

__Actual behavior:__

I get the last entry written in the day before

__Environment info:__

* System info: Linux 4.19.0-14-amd64 x86_64
* InfluxDB version: InfluxDB 2.0.3 (git: fe04d346df) build_date: 2020-12-15T01:00:16Z* Other

**Comment by danxmoran** [comment]
Could be related to #20595

**Comment by danxmoran** [comment]
Possible duplicate of influxdata/flux#3480

**Comment by danxmoran** [comment]
Fixed by #21140
